platform/search/test/controllers/SearchMenuControllerSpec.js

```diff
@@ -83,7 +83,7 @@ define(
                 mockScope.ngModel.checkAll = false;
 
                 controller.checkAll();
-                expect(mockScope.ngModel.filtersString).toEqual('NONE');
```

Comment:
The description of this test is a little inconsistent with its implementation now. I wonder if we should repurpose it to test that the "All" option cannot be unset when nothing is selected.

I'll leave it up to you how best to implement this. It could be as simple as taking the existing test and just adding some lines to verify that `ngModel.checkAll` is still true after each invocation of `controller.checkAll()`. You might also want to add a check on the precondition that nothing is selected.